lib/Player.rb

```diff
@@ -0,0 +1,34 @@
+class Player
+    attr_accessor :name, :piece
+    def initialize(name, piece, board)
+        @name = name 
+        @piece = piece
+        @board = board
+    end
+
+    def ask_move
+        loop do
+            puts "Enter move to make "
+            move = gets.chomp
+            cordinate = map_cordinates(move)
+            break if @board.add_piece(cordinate, @piece)
+        end
+    end
+
+    private
+    
+        def map_cordinates(number)
```

Comment:
What does this map_coordinates method do?

Reply:
The map_coordinates method converts our user's input from an integer to a coordinate in our two dimensional array for example it maps 9 to [2,2] which indicates the row and column where 9 is found on our two dimensional array.

Reply:
What exactly does that accomplish in relation to what the Player should be doing? At any rate is there a way we can just perform the same thing with just using an array instead of a hash? For instance instead of having a multidimensional storage you could just store the squares in your board as a single dimension array: [0, 1, 2, 3, 4, 5, 6, 7, 8]. Which is represented visually as such:

```
0  1  2
3  4  5
6  7  8
```
Your methods will be much simpler and easier to maintain if you use simpler data types.